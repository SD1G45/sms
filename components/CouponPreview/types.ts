export interface CouponPreviewProps {
  title: string;
  description: string;
  primaryColor: string;
  expirationDate: Date;
}
