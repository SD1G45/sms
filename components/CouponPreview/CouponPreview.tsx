import React from "react";
import Timer from "../Timer";
import {
  Phone,
  Heading,
  Container,
  Notch,
  BusinessLogo,
  Description,
  Title,
  RedeemButton,
} from "./styles";
import { CouponPreviewProps } from "./types";

const CouponPreview: React.FC<CouponPreviewProps> = ({
  primaryColor,
  title,
  description,
  expirationDate,
}) => {
  return (
    <Container>
      <Heading>Phone preview</Heading>
      <Phone>
        <Notch />
        <BusinessLogo primaryColor={primaryColor} />
        <Title>{title}</Title>
        <Description>{description}</Description>
        <RedeemButton primaryColor={primaryColor}>Redeem</RedeemButton>
        <Timer expirationDate={expirationDate} />
      </Phone>
    </Container>
  );
};

export default CouponPreview;
