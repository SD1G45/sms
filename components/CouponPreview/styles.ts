import styled from "styled-components";
import { CouponPreviewProps } from "./types";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Heading = styled.h2`
  font-size: 1rem;
  margin-bottom: 30px;
`;

export const Phone = styled.div`
  height: 700px;
  border-radius: 20px;
  width: 350px;
  border: 2px solid ${(props) => props.theme.colors.lightGray};
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 60px 20px;
`;

export const Notch = styled.div`
  position: absolute;
  top: -2px;
  left: 100px;
  width: 150px;
  height: 20px;
  background-color: white;
  border-radius: 0 0 10px 10px;
  border: 2px solid ${(props) => props.theme.colors.lightGray};
  border-top: 0;
`;

export const BusinessLogo = styled.img<
  Pick<CouponPreviewProps, "primaryColor">
>`
  border-radius: 100%;
  width: 100px;
  height: 100px;
  border: 4px solid ${(props) => props.primaryColor};
  margin-bottom: 40px;
`;

export const Title = styled.h1`
  font-size: 1rem;
  margin-bottom: 10px;
  text-align: center;
`;

export const Description = styled.p`
  font-size: 1rem;
  margin-bottom: 40px;
  text-align: center;
`;

export const RedeemButton = styled.button<
  Pick<CouponPreviewProps, "primaryColor">
>`
  width: 150px;
  height: 150px;
  color: white;
  background-color: ${(props) => props.primaryColor};
  border-radius: 100%;
`;
