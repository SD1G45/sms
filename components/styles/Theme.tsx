import React from "react";
import { ThemeProvider } from "styled-components";
import ITheme from "./types";

const screenWidth = {
  minSmall: `(min-width: ${430}px)`,
  minMedium: `(min-width: ${730}px)`,
  minLarge: `(min-width: ${1024}px)`,
  maxSmall: `(max-width: ${430}px)`,
  maxMedium: `(max-width: ${730}px)`,
  maxLarge: `(max-width: ${1024}px)`,
};

const lightTheme: ITheme = {
  colors: {
    primary: "#4881F0",
    lightGray: "#BEC8D2",
    textPrimary: "#000000",
    error: "#FF7F7F",
  },
  screenWidth: { ...screenWidth },
};

interface ThemeProps {
  children: React.ReactNode;
}

const Theme: React.FC<ThemeProps> = ({ children }) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);

export default Theme;
