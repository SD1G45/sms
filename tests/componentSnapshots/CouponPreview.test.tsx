import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import CouponPreview from "../../components/CouponPreview";
import Theme from "../../components/styles/Theme";

test("Coupon preview renders correctly", () => {
  const tree = renderer
    .create(
      <Theme>
        <CouponPreview
          title={"Coupon"}
          description={"This is a sample coupon"}
          expirationDate={new Date(0)}
          primaryColor={"#fff"}
        />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});