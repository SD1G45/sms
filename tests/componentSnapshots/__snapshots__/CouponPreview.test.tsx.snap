// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon preview renders correctly 1`] = `
.c10 {
  text-color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 40%;
}

.c8 {
  width: 300px;
  padding: 30px;
  text-align: center;
}

.c13 {
  text-align: left;
}

.c11 {
  margin: 10px;
}

.c12 {
  border-radius: 30%;
  border: 2px solid black;
  margin-right: 5px;
  width: 40px;
  height: 40px;
  display: table-cell;
  vertical-align: middle;
  border-color: black;
}

.c9 {
  text-align: center;
  opacity: 40%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 1rem;
  margin-bottom: 30px;
}

.c2 {
  height: 700px;
  border-radius: 20px;
  width: 350px;
  border: 2px solid #BEC8D2;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 60px 20px;
}

.c3 {
  position: absolute;
  top: -2px;
  left: 100px;
  width: 150px;
  height: 20px;
  background-color: white;
  border-radius: 0 0 10px 10px;
  border: 2px solid #BEC8D2;
  border-top: 0;
}

.c4 {
  border-radius: 100%;
  width: 100px;
  height: 100px;
  border: 4px solid #fff;
  margin-bottom: 40px;
}

.c5 {
  font-size: 1rem;
  margin-bottom: 10px;
  text-align: center;
}

.c6 {
  font-size: 1rem;
  margin-bottom: 40px;
  text-align: center;
}

.c7 {
  width: 150px;
  height: 150px;
  color: white;
  background-color: #fff;
  border-radius: 100%;
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Phone preview
  </h2>
  <div
    className="c2"
  >
    <div
      className="c3"
    />
    <img
      className="c4"
    />
    <h1
      className="c5"
    >
      Coupon
    </h1>
    <p
      className="c6"
    >
      This is a sample coupon
    </p>
    <button
      className="c7"
    >
      Redeem
    </button>
    <div
      className="c8"
    >
      <div
        className="c9"
      >
        Valid for the next 
      </div>
      <div
        className="c10"
      >
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            0
          </div>
          <p
            className="c13"
          >
            Days
          </p>
        </div>
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            0
          </div>
          <p
            className="c13"
          >
            Hours
          </p>
        </div>
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            0
          </div>
          <p
            className="c13"
          >
            Mins
          </p>
        </div>
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            0
          </div>
          <p
            className="c13"
          >
            Secs
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
